### PR TITLE
Update API to v2

### PR DESCRIPTION
**Pull request by abdyfranco**

### CODE REVIEW

File: apis/commands/vultr_instances.php [hunk 1]
```diff
@@ -0,0 +1,689 @@
...
+    /**
+     * Halt a virtual machine.
+     *
+     * @param array $params An array containing the following arguments:
+     *     - instance_ids: Unique identifier for this instance.
+     * @return VultrResponse An object containing the api response.
+     */
+    public function halt($params = [])
+    {
+        return $this->api->apiRequest('/instances/halt', $params, 'POST');
```
Comment (JReissmueller) [inline apis/commands/vultr_instances.php]:
```
return $this->api->apiRequest('/instances/' . ($params['instance-id'] ?? '') . '/halt', $params, 'POST');
```

File: apis/commands/vultr_instances.php [hunk 2]
```diff
@@ -0,0 +1,689 @@
...
+    /**
+     * Reboot a virtual machine.
+     *
+     * @param array $params An array containing the following arguments:
+     *     - instance-ids: The Instance IDs to reboot.
+     * @return VultrResponse An object containing the api response.
+     */
+    public function reboot($params = [])
+    {
+        return $this->api->apiRequest('/instances/reboot', $params, 'POST');
```
Comment (JReissmueller) [inline apis/commands/vultr_instances.php]:
```
return $this->api->apiRequest('/instances/' . ($params['instance-id'] ?? '') . '/reboot', $params, 'POST');
```

File: apis/commands/vultr_instances.php [hunk 3]
```diff
@@ -0,0 +1,689 @@
...
+    /**
+     * Start a virtual machine. If the machine is already running, it will be restarted.
+     *
+     * @param array $params An array containing the following arguments:
+     *     - instance-id: Unique identifier for this instance.
+     * @return VultrResponse An object containing the api response.
+     */
+    public function start($params = [])
+    {
+        return $this->api->apiRequest('/instances/start', $params, 'POST');
```
Comment (JReissmueller) [inline apis/commands/vultr_instances.php]:
```
return $this->api->apiRequest('/instances/' . ($params['instance-id'] ?? '') . '/start', $params, 'POST');
```

File: vultr.php [hunk 4]
```diff
@@ -2128,28 +2148,28 @@ private function getTabActions($package, $service, array $post = null, $client =
...
-                            'SUBID' => $service_fields->vultr_subid,
+                            $instance_key => $service_fields->vultr_subid,
                             'hostname' => $service_fields->vultr_hostname
                         ];
+
+                        if ($package->meta->server_type !== 'server') {
+                            unset($params['hostname']);
+                        }
+
                         $vultr_api->reinstall($params);
```
Comment (JReissmueller) [inline vultr.php]:
```
$this->parseResponse($vultr_api->reinstall($params));
```

File: vultr.php [hunk 5]
```diff
@@ -2128,28 +2148,28 @@ private function getTabActions($package, $service, array $post = null, $client =
             } else {
                 switch ($post['action']) {
                     case 'restart':
+                    case 'start':
                         $params = [
-                            'SUBID' => $service_fields->vultr_subid
+                            $instance_key => $service_fields->vultr_subid
                         ];
                         $vultr_api->reboot($params);
```
Comment (JReissmueller) [inline vultr.php]:
```
$this->parseResponse($vultr_api->reboot($params));
```

File: vultr.php [hunk 6]
```diff
@@ -2128,28 +2148,28 @@ private function getTabActions($package, $service, array $post = null, $client =
...
                         break;
                     case 'stop':
                         $params = [
-                            'SUBID' => $service_fields->vultr_subid
+                            $instance_key => $service_fields->vultr_subid
                         ];
                         $vultr_api->halt($params);
```
Comment (JReissmueller) [inline vultr.php]:
```
$this->parseResponse($vultr_api->halt($params));
```

File: apis/commands/vultr_baremetal.php [hunk 7]
```diff
@@ -28,237 +28,273 @@ public function __construct(VultrApi $api)
...
-     *     - enable_ipv6: 'yes' or 'no'.  If yes, an IPv6 subnet will be assigned to the server.
-     *     - label: This is a text label that will be shown in the control panel.
-     *     - SSHKEYID: List of SSH keys to apply to this server on install (only valid for Linux/FreeBSD).
-     *     - APPID: If launching an application (OSID 186), this is the app id to launch.
-     *     - userdata: Base64 encoded user-data.
-     *     - notify_activate: 'yes' or 'no'. If yes, an email will be sent when the server is ready.
-     *     - hostname: The hostname to assign to this server.
-     *     - tag: The tag to assign to this server.
+     * @param array $params An array containing the following arguments:
+     *     - region: The Region id to create the instance.
```
Comment (JReissmueller) [inline apis/commands/vultr_baremetal.php:32]: This list should also include plan, label, hostname, enable_ipv6, and app_id

File: apis/commands/vultr_baremetal.php [hunk 8]
```diff
@@ -28,237 +28,273 @@ public function __construct(VultrApi $api)
...
-     * @param array $params An array contaning the following arguments:
-     *     - SUBID: Unique identifier for this subscription.
+     * @param array $params An array containing the following arguments:
+     *     - baremetal_ids: The Bare Metal id.
      * @return VultrResponse An instance of the API response
      */
     public function halt($params = [])
     {
-        return $this->api->apiRequest('/baremetal/halt', $params, 'POST');
+        return $this->api->apiRequest('/bare-metals/halt', $params, 'POST');
```
Comment (JReissmueller) [inline apis/commands/vultr_baremetal.php]: return $this->api->apiRequest('/bare-metals/' . ($params['baremetal-id'] ?? '') . '/halt', $params, 'POST');

File: apis/commands/vultr_baremetal.php [hunk 9]
```diff
@@ -28,237 +28,273 @@ public function __construct(VultrApi $api)
...
-     * @param array $params An array contaning the following arguments:
-     *     - SUBID: Unique identifier of a subscription.
+     * @param array $params An array containing the following arguments:
+     *     - baremetal_ids: The Bare Metal id.
      * @return VultrResponse An instance of the API response
      */
     public function reboot($params = [])
     {
-        return $this->api->apiRequest('/baremetal/reboot', [], 'POST');
+        return $this->api->apiRequest('/bare-metals/reboot', $params, 'POST');
```
Comment (JReissmueller) [inline apis/commands/vultr_baremetal.php]: return $this->api->apiRequest('/bare-metals/' . ($params['baremetal-id'] ?? '') . '/reboot', $params, 'POST');

File: vultr.php [hunk 10]
```diff
@@ -29,6 +29,259 @@ public function __construct()
...
+     * Performs migration of data from $current_version (the current installed version)
+     * to the given file set version. Sets Input errors on failure, preventing
+     * the module from being upgraded.
+     *
+     * @param string $current_version The current installed version of this module
+     */
+    public function upgrade($current_version)
+    {
+        // Upgrade to 2.0.0
+        if (version_compare($current_version, '2.0.0', '<')) {
```
Comment (JReissmueller) [inline vultr.php:42]: I tried creating a service on the previous version, then updating. That service was not properly updated.  It was a VM, not a baremetal.

**Review by JReissmueller — CHANGES_REQUESTED** [review]

File: vultr.php (hunk 10 above)
Comment (abdyfranco) [reply vultr.php:42]: There was a specific error or unexpected behavior?

Comment (JReissmueller) [reply vultr.php:42]: Sorry, I was vague.  The subid service field was not updated and thus none of the management tabs or api calls functioned.

Comment (abdyfranco) [reply vultr.php:42]: I was not able to reproduce the issue, but I made some updates to improve the upgrade process.

**Review by JReissmueller — APPROVED** [review]

**Pull request merged**